src/Support/Traits/DataContainerTrait.php

```diff
@@ -118,7 +118,11 @@ public function has($key)
      */
     public function forget($key)
     {
-        Arr::forget($this->items, $key);
+        $items = $this->items;
+
+        Arr::forget($items, $key);
+
+        $this->items = $items;
```

Comment:
It doesn't seem like it, dumping the items array after it's set, it's the same array. At least in PHP `v5.6`